Release 1.15.0 (#5921)
antrea-io/antrea: CHANGELOG/CHANGELOG-1.15.md

# Changelog 1.15

## 1.15.0 - 2024-01-25

### Added

> - Add a new feature `NodeNetworkPolicy` to allow user to apply `ClusterNetworkPolicy` to Kubernetes Nodes. ([#5658](https://github.com/antrea-io/antrea/pull/5658) [#5716](https://github.com/antrea-io/antrea/pull/5716), [@Atish-iaf])

Reviewer: ```suggestion
- Add a new feature `NodeNetworkPolicy` to allow user to apply `ClusterNetworkPolicy` to Kubernetes Nodes. ([#5658](https://github.com/antrea-io/antrea/pull/5658) [#5716](https://github.com/antrea-io/antrea/pull/5716), [@hongliangl] [@Atish-iaf])
```

Author: Fixed.

> - Add a new feature `NodeNetworkPolicy` to allow user to apply `ClusterNetworkPolicy` to Kubernetes Nodes. ([#5658](https://github.com/antrea-io/antrea/pull/5658) [#5716](https://github.com/antrea-io/antrea/pull/5716), [@hongliangl] [@Atish-iaf])

Reviewer 2: Can we add doc links for the well-documented features? This may help users get started quickly。
I think the first 4 are well documented.

Author: Done.

> - Add L7 network flow export support in Antrea to enables users to export L7 protocol information. ([#5218](https://github.com/antrea-io/antrea/pull/5218), [@tushartathgur])

Reviewer 3: "to enables" to "to enable"

> - Add migration tool to support migrating from other CNI to Antrea. ([#5677](https://github.com/antrea-io/antrea/pull/5677), [@hjiajing])

Reviewer: ```suggestion
- Add migration tool to support migrating from other CNIs to Antrea. ([#5677](https://github.com/antrea-io/antrea/pull/5677), [@hjiajing])
```

- Support Egress using IPs from a separate subnet through VLAN. ([#5799](https://github.com/antrea-io/antrea/pull/5799), [@tnqn])

> - Record an event when Egress IP assignment changes for better troubleshooting. ([#5765](https://github.com/antrea-io/antrea/pull/5765), [@jainpulkit22])

Author: Moded

> - Add Support for `FlexibleIPAM` on Antrea Multicast so that multicast packets are forwarded to the uplink interface when `FlexibleIPAM` is enabled. ([#4922](https://github.com/antrea-io/antrea/pull/4922), [@ceclinux])

Reviewer: ```suggestion
- Add Support for `FlexibleIPAM` on Antrea Multicast so that multicast packets can be forwarded to the uplink interface when `FlexibleIPAM` is enabled. ([#4922](https://github.com/antrea-io/antrea/pull/4922), [@ceclinux])
```

Author: updated.

> - Add Support for `FlexibleIPAM` on Antrea Multicast so that multicast packets can be forwarded to the uplink interface when `FlexibleIPAM` is enabled. ([#4922](https://github.com/antrea-io/antrea/pull/4922), [@ceclinux])

Reviewer 2: We don't have a feature gate called FlexibleIPAM, and the words starting with "so that" is perhaps too implementation specific.
Perhaps just say "Add Antrea flexible IPAM support for the Multicast feature"

Author: Updated

- Support Talos clusters to run Antrea as the CNI, and add Talos to the K8s installers document. ([#5718](https://github.com/antrea-io/antrea/pull/5718) [#5766](https://github.com/antrea-io/antrea/pull/5766), [@antoninbas])

### Changed

- Support secondary network when the network configuration in `NetworkAttachmentDefinition` does not include IPAM configuration. ([#5762](https://github.com/antrea-io/antrea/pull/5762), [@jianjuns])
- Change secondary network Pod controller to subscribe to CNIServer events to support bridging and VLAN network. ([#5767](https://github.com/antrea-io/antrea/pull/5767), [@jianjuns])
- Use Antrea IPAM for Antrea native secondary network support. ([#5427](https://github.com/antrea-io/antrea/pull/5427), [@jianjuns])

> - Build different images for antrea-agent and antrea-controller to reduce the image size, which can accelerate the antrea-agent and antrea-controller startup. ([#5856](https://github.com/antrea-io/antrea/pull/5856) [#5902](https://github.com/antrea-io/antrea/pull/5902) [#5903](https://github.com/antrea-io/antrea/pull/5903), [@jainpulkit22])

Reviewer 3: Might prefer like "Create different images for antrea-agent and antrea-controller to minimize the overall image size, speeding up the startup of both antrea-agent and antrea-controller"

Author: Refined.

- Update Windows documentation with more clearer installation guide and instructions. ([#5789](https://github.com/antrea-io/antrea/pull/5789), [@antoninbas])
- Enable IPv4/IPv6 forwarding on demand automatically to eliminate the need for user intervention or dependencies on other components. ([#5833](https://github.com/antrea-io/antrea/pull/5833), [@tnqn])
- Prioritize L7 NetworkPolicy flows over `TrafficControl` to avoid a potential issue that a `TrafficControl` CR with a redirect action to the same Pod could bypass the L7 engine. ([#5768](https://github.com/antrea-io/antrea/pull/5768), [@hongliangl])
- Add ability to skip loading kernel modules in antrea-agent to support some specialized distributions (eg: Talos). ([#5754](https://github.com/antrea-io/antrea/pull/5754), [@antoninbas])
- Add NetworkPolicy rule name in Traceflow observation. ([#5667](https://github.com/antrea-io/antrea/pull/5667), [@Atish-iaf])
- Use Traceflow API v1beta1 to replace the deprecated API version in `antctl traceflow`. ([#5689](https://github.com/antrea-io/antrea/pull/5689), [@Atish-iaf])

> - Replace `net.IP` with `netip.Addr` in FlowExporter. ([#5532](https://github.com/antrea-io/antrea/pull/5532), [@antoninbas])

Reviewer 2: It should mention the optimiation of memory usage, otherwise the change looks meaningless to users

Author: Done.

- Update kubemark from v1.18.4 to v1.29.0 for antrea-agent-simulator. ([#5820](https://github.com/antrea-io/antrea/pull/5820), [@luolanzone])
- Update Antrea ROADMAP.md document. ([#5807](https://github.com/antrea-io/antrea/pull/5807), [@antoninbas])
- Upgrade CNI plugins to v1.4.0. ([#5747](https://github.com/antrea-io/antrea/pull/5747) [#5813](https://github.com/antrea-io/antrea/pull/5813), [@antoninbas] [@luolanzone])

> - Upgrade Prometheus image to v2.46.0. ([#5023](https://github.com/antrea-io/antrea/pull/5023), [@luolanzone])

Reviewer 2: I feel these version updates are not worth to mention

- Update the document for Egress feature's options and usage on AWS cloud. ([#5436](https://github.com/antrea-io/antrea/pull/5436), [@tnqn])
- Add instructions to install Antrea in `encap` mode in AKS. ([#5901](https://github.com/antrea-io/antrea/pull/5901), [@antoninbas])
- Add Flexible IPAM design details in `antrea-ipam.md`. ([#5339](https://github.com/antrea-io/antrea/pull/5339), [@gran-vmv])

### Fixed

> 

Reviewer 2: https://github.com/antrea-io/antrea/pull/5880 should be mentioned

Author: Added this one.

- Store NetworkPolicy in filesystem as fallback data source to let antre-agent fallback to use the files if it can't connect to antrea-controller on startup. ([#5739](https://github.com/antrea-io/antrea/pull/5739), [@tnqn])
- Enable Pod network after realizing initial NetworkPolicies to avoid traffic from/to Pods bypassing NetworkPolicy when antrea-agent restarts. ([#5777](https://github.com/antrea-io/antrea/pull/5777), [@tnqn])
- Fix Clean-AntreaNetwork.ps1 invocation in Prepare-AntreaAgent.ps1 for containerized OVS on Windows. ([#5859](https://github.com/antrea-io/antrea/pull/5859), [@antoninbas])
- Add missing space to kubelet args in Prepare-Node.ps1 so that kubelet can start successfully on Windows. ([#5858](https://github.com/antrea-io/antrea/pull/5858), [@antoninbas])
- Fix `antctl trace-packet` command failure which is caused by arguments missing issue. ([#5838](https://github.com/antrea-io/antrea/pull/5838), [@luolanzone])
- Support Local ExternalTrafficPolicy for Services with ExternalIPs when Antrea proxyAll mode is enabled. ([#5795](https://github.com/antrea-io/antrea/pull/5795), [@tnqn])

> - Set net.ipv4.conf.antrea-gw0.arp_announce to 1 to fix an ARP request leak when a Node or hostNetwork Pod accesses a local Pod and AntreaIPAM is enabled. ([#5657](https://github.com/antrea-io/antrea/pull/5657), [@gran-vmv])

Reviewer: ```suggestion
- Set `net.ipv4.conf.antrea-gw0.arp_announce` to 1 to fix an ARP request leak when a Node or hostNetwork Pod accesses a local Pod and AntreaIPAM is enabled. ([#5657](https://github.com/antrea-io/antrea/pull/5657), [@gran-vmv])
```

- Support secondary network when the network configuration in `NetworkAttachmentDefinition` does not include IPAM configuration. ([#5762](https://github.com/antrea-io/antrea/pull/5762), [@jianjuns])
- Add instructions to install Antrea in `encap` mode in AKS. ([#5901](https://github.com/antrea-io/antrea/pull/5901), [@antoninbas])

### Changed

- Change secondary network Pod controller to subscribe to CNIServer events to support bridging and VLAN network. ([#5767](https://github.com/antrea-io/antrea/pull/5767), [@jianjuns])
- Use Antrea IPAM for secondary network support. ([#5427](https://github.com/antrea-io/antrea/pull/5427), [@jianjuns])
- Create different images for antrea-agent and antrea-controller to minimize the overall image size, speeding up the startup of both antrea-agent and antrea-controller. ([#5856](https://github.com/antrea-io/antrea/pull/5856) [#5902](https://github.com/antrea-io/antrea/pull/5902) [#5903](https://github.com/antrea-io/antrea/pull/5903), [@jainpulkit22])

> - Fix incorrect MTU configurations for the WireGuard encryption mode and GRE tunnel mode. ([#5880](https://github.com/antrea-io/antrea/pull/5880) [#5926](https://github.com/antrea-io/antrea/pull/5926), [@hjiajing] [@tnqn])

Reviewer 4: This should be "Fixed"?

Reviewer 2: Yes, this is a bugfix

Author: Moved

- Don't create tunnel interface (antrea-tun0) when using Wireguard encryption mode. ([#5885](https://github.com/antrea-io/antrea/pull/5885) [#5909](https://github.com/antrea-io/antrea/pull/5909), [@antoninbas])
- Record an event when Egress IP assignment changes for better troubleshooting. ([#5765](https://github.com/antrea-io/antrea/pull/5765), [@jainpulkit22])

> - Delete OVS port and flows before releasing Pod IP. ([#5788](https://github.com/antrea-io/antrea/pull/5788), [@tnqn])

Reviewer 4: Should be Fixed? @tnqn

Reviewer 2: Yes

- Update Windows documentation with clearer installation guide and instructions. ([#5789](https://github.com/antrea-io/antrea/pull/5789), [@antoninbas])
- Enable IPv4/IPv6 forwarding on demand automatically to eliminate the need for user intervention or dependencies on other components. ([#5833](https://github.com/antrea-io/antrea/pull/5833), [@tnqn])

> - Prioritize L7 NetworkPolicy flows over `TrafficControl` to avoid a potential issue that a `TrafficControl` CR with a redirect action to the same Pod could bypass the L7 engine. ([#5768](https://github.com/antrea-io/antrea/pull/5768), [@hongliangl])

Reviewer 4: Should this be Fixed @tnqn ?

Reviewer 2: Yes

# Changelog 1.15

## 1.15.0 - 2024-01-26

### Added

- Add a new feature `NodeNetworkPolicy` to allow user to apply `ClusterNetworkPolicy` to Kubernetes Nodes. ([#5658](https://github.com/antrea-io/antrea/pull/5658) [#5716](https://github.com/antrea-io/antrea/pull/5716), [@hongliangl] [@Atish-iaf])
    * Refer to [this document](https://github.com/antrea-io/antrea/blob/release-1.15/docs/antrea-node-network-policy.md) for more information about this feature.

> - Add L7 network flow export support in Antrea to enable users to export L7 protocol information. ([#5218](https://github.com/antrea-io/antrea/pull/5218), [@tushartathgur])

Author: Updated.

    * Refer to [this document](https://github.com/antrea-io/antrea/blob/release-1.15/docs/network-flow-visibility.md) for more information about this feature.
- Add migration tool to support migrating from other CNIs to Antrea. ([#5677](https://github.com/antrea-io/antrea/pull/5677), [@hjiajing])
    * Refer to [this document](https://github.com/antrea-io/antrea/blob/release-1.15/docs/migrate-to-antrea.md) for more information about this tool.

> - Support Egress using IPs from a different subnet from the default Node subnet. ([#5799](https://github.com/antrea-io/antrea/pull/5799), [@tnqn])

Reviewer 5: s/from a different subnet from the default Node subnet/from a subnet that is different from the default Node subnet

Author: Updated.

    * Refer to [this document](https://github.com/antrea-io/antrea/blob/release-1.15/docs/egress.md) for more information about this feature.
- Add Antrea flexible IPAM support for the Multicast feature. ([#4922](https://github.com/antrea-io/antrea/pull/4922), [@ceclinux])
- Support Talos clusters to run Antrea as the CNI, and add Talos to the K8s installers document. ([#5718](https://github.com/antrea-io/antrea/pull/5718) [#5766](https://github.com/antrea-io/antrea/pull/5766), [@antoninbas])

### Changed

> - Support secondary network when the network configuration in `NetworkAttachmentDefinition` does not include IPAM configuration. ([#5762](https://github.com/antrea-io/antrea/pull/5762), [@jianjuns])

Reviewer 4: I feel this one can be moved to "Added".

- Change secondary network Pod controller to subscribe to CNIServer events to support bridging and VLAN network. ([#5767](https://github.com/antrea-io/antrea/pull/5767), [@jianjuns])

> - Use Antrea IPAM for Antrea native secondary network support. ([#5427](https://github.com/antrea-io/antrea/pull/5427), [@jianjuns])

Reviewer 5: remove "Antrea native" as you don't specify it for the 2 previous bullet points

- Create different images for antrea-agent and antrea-controller to minimize the overall image size, speeding up the startup of both antrea-agent and antrea-controller. ([#5856](https://github.com/antrea-io/antrea/pull/5856) [#5902](https://github.com/antrea-io/antrea/pull/5902) [#5903](https://github.com/antrea-io/antrea/pull/5903), [@jainpulkit22])

> - Fix incorrect MTU configurations for the WireGuard encryption mode and GRE tunnel mode. ([#5880](https://github.com/antrea-io/antrea/pull/5880), [@hjiajing] [@tnqn])

Reviewer 2: ```suggestion
- Fix incorrect MTU configurations for the WireGuard encryption mode and GRE tunnel mode. ([#5880](https://github.com/antrea-io/antrea/pull/5880) [#5926](https://github.com/antrea-io/antrea/pull/5926), [@hjiajing] [@tnqn])
```

- Don't create tunnel interface (antrea-tun0) when using Wireguard encryption mode. ([#5885](https://github.com/antrea-io/antrea/pull/5885) [#5909](https://github.com/antrea-io/antrea/pull/5909), [@antoninbas])
- Record an event when Egress IP assignment changes for better troubleshooting. ([#5765](https://github.com/antrea-io/antrea/pull/5765), [@jainpulkit22])
- Delete OVS port and flows before releasing Pod IP. ([#5788](https://github.com/antrea-io/antrea/pull/5788), [@tnqn])

> - Update Windows documentation with more clearer installation guide and instructions. ([#5789](https://github.com/antrea-io/antrea/pull/5789), [@antoninbas])

Reviewer 5: s/more clearer/clearer

- Enable IPv4/IPv6 forwarding on demand automatically to eliminate the need for user intervention or dependencies on other components. ([#5833](https://github.com/antrea-io/antrea/pull/5833), [@tnqn])
- Prioritize L7 NetworkPolicy flows over `TrafficControl` to avoid a potential issue that a `TrafficControl` CR with a redirect action to the same Pod could bypass the L7 engine. ([#5768](https://github.com/antrea-io/antrea/pull/5768), [@hongliangl])

> - Add ability to skip loading kernel modules in antrea-agent to support some specialized distributions (eg: Talos). ([#5754](https://github.com/antrea-io/antrea/pull/5754), [@antoninbas])

Reviewer 5: nit: s/eg:/e.g.,

- Add NetworkPolicy rule name in Traceflow observation. ([#5667](https://github.com/antrea-io/antrea/pull/5667), [@Atish-iaf])

> - Use Traceflow API v1beta1 to replace the deprecated API version in `antctl traceflow`. ([#5689](https://github.com/antrea-io/antrea/pull/5689), [@Atish-iaf])

Reviewer 5: s/to replace/instead of

> - Replace `net.IP` with `netip.Addr` in FlowExporter which optimizes the memory usage and improve the performance of the FlowExporter. ([#5532](https://github.com/antrea-io/antrea/pull/5532), [@antoninbas])

Reviewer 5: s/improve/improves

- Update kubemark from v1.18.4 to v1.29.0 for antrea-agent-simulator. ([#5820](https://github.com/antrea-io/antrea/pull/5820), [@luolanzone])

> - Update Antrea ROADMAP.md document. ([#5807](https://github.com/antrea-io/antrea/pull/5807), [@antoninbas])

Reviewer 5: Maybe we don't need to mention this in release notes?

Author: removed

- Upgrade CNI plugins to v1.4.0. ([#5747](https://github.com/antrea-io/antrea/pull/5747) [#5813](https://github.com/antrea-io/antrea/pull/5813), [@antoninbas] [@luolanzone])
- Update the document for Egress feature's options and usage on AWS cloud. ([#5436](https://github.com/antrea-io/antrea/pull/5436), [@tnqn])

> - Add instructions to install Antrea in `encap` mode in AKS. ([#5901](https://github.com/antrea-io/antrea/pull/5901), [@antoninbas])

Reviewer 4: Maybe move this one to Added.

Reviewer 4: So, you agree to move this one to Added or not?

Author: Forgot to update here, it's already moved.

- Add Flexible IPAM design details in `antrea-ipam.md`. ([#5339](https://github.com/antrea-io/antrea/pull/5339), [@gran-vmv])

### Fixed

- Store NetworkPolicy in filesystem as fallback data source to let antre-agent fallback to use the files if it can't connect to antrea-controller on startup. ([#5739](https://github.com/antrea-io/antrea/pull/5739), [@tnqn])
- Enable Pod network after realizing initial NetworkPolicies to avoid traffic from/to Pods bypassing NetworkPolicy when antrea-agent restarts. ([#5777](https://github.com/antrea-io/antrea/pull/5777), [@tnqn])
- Fix Clean-AntreaNetwork.ps1 invocation in Prepare-AntreaAgent.ps1 for containerized OVS on Windows. ([#5859](https://github.com/antrea-io/antrea/pull/5859), [@antoninbas])
- Add missing space to kubelet args in Prepare-Node.ps1 so that kubelet can start successfully on Windows. ([#5858](https://github.com/antrea-io/antrea/pull/5858), [@antoninbas])

> - Fix `antctl trace-packet` command failure which is caused by arguments missing issue. ([#5838](https://github.com/antrea-io/antrea/pull/5838), [@luolanzone])

Reviewer 5: s/caused by arguments missing issue/caused by missing arguments

- Support Local ExternalTrafficPolicy for Services with ExternalIPs when Antrea proxyAll mode is enabled. ([#5795](https://github.com/antrea-io/antrea/pull/5795), [@tnqn])
- Set `net.ipv4.conf.antrea-gw0.arp_announce` to 1 to fix an ARP request leak when a Node or hostNetwork Pod accesses a local Pod and AntreaIPAM is enabled. ([#5657](https://github.com/antrea-io/antrea/pull/5657), [@gran-vmv])
- Skip enforcement of ingress NetworkPolicies rules for hairpinned Service traffic (Pod accessing itself via a Service). ([#5687](https://github.com/antrea-io/antrea/pull/5687) [#5705](https://github.com/antrea-io/antrea/pull/5705), [@GraysonWu])
- Add host-local IPAM GC on startup to avoid potential IP leak issue after antrea-agent restart. ([#5660](https://github.com/antrea-io/antrea/pull/5660), [@antoninbas])

> - Fix antrea-ubi image build failure with ubi8 image. ([#5723](https://github.com/antrea-io/antrea/pull/5723), [@antoninbas])

Reviewer 5: this doesn't match the changelog entry we already have for 1.14.1:
```
$ git grep 5723
CHANGELOG-1.14.md:- Fix the CrashLookBackOff issue when using the UBI-based image. ([#5723](https://github.com/antrea-io/antrea/pull/5723), [@antoninbas])
```

Author: Fixed.

- Remove redundant log in `fillPodInfo`/`fillServiceInfo` to fix log flood issue, and update `DestinationServiceAddress` for deny connections. ([#5592](https://github.com/antrea-io/antrea/pull/5592) [#5704](https://github.com/antrea-io/antrea/pull/5704), [@yuntanghsu])
- Enhance HNS network initialization on Windows to avoid some corner cases. ([#5841](https://github.com/antrea-io/antrea/pull/5841), [@XinShuYang])
- Fix endpoint querier rule index in response to improve troubleshooting. ([#5783](https://github.com/antrea-io/antrea/pull/5783), [@qiyueyao])

> - Enhance ExternalIPPool validation to avoid unexpected behaviors. ([#5898](https://github.com/antrea-io/antrea/pull/5898), [@aroradaman])

Reviewer 2: As it hasn't been merged, we should remove it if there is no update today.

Author: Will remove this later if there is still no change for the PR.

Author: Removed this considering it's not critical and might not be merged in time.

> - Fix redundant reconciles in FQDN controller to avoid unnecessary reconciliations. ([#5893](https://github.com/antrea-io/antrea/pull/5893), [@Dyanngg])

Reviewer 5: suggesting something more concise for this 1:
> Avoid unnecessary rule reconciliations in FQDN controller

> - Update Windows OVS download link to remove the redundant certificate to fix unsigned OVS Windows driver. ([#5839](https://github.com/antrea-io/antrea/pull/5839), [@XinShuYang])

Reviewer 5: s/to remove the redundant certificate to fix unsigned OVS Windows driver/to remove the invalid certificate preventing unsigned OVS driver installation

> - Fix IP annotation not working on a StatefulSet for Antrea FlexibleIPAM. ([#5715](https://github.com/antrea-io/antrea/pull/5715), [@gran-vmv])

Reviewer 5: s/on a StatefulSet/on StatefulSets

> - Add DHCP IP retries logic in `PrepareHNSNetwork` to fix potential IP retrieving failure. ([#5819](https://github.com/antrea-io/antrea/pull/5819), [@XinShuYang])

Reviewer 5: nit: remove logic
